File: rpg/schemas/game.py
"""schemas.game — 游戏核心流程路由请求模型。"""
from __future__ import annotations
from typing import Optional, Any
from schemas._common import _BaseRequest


class NewGameRequest(_BaseRequest):
    script_card_id: Optional[Any] = None
    script_id: Optional[Any] = None
    user_card_id: Optional[Any] = None
    persona_id: Optional[Any] = None
    role: Optional[str] = ""
    name: Optional[str] = "无名者"
    background: Optional[str] = ""


class ChatEstimateRequest(_BaseRequest):
    message: Optional[str] = ""
    include_retrieval: Optional[bool] = True


class ChatRequest(_BaseRequest):
    message: Optional[str] = ""
    text: Optional[str] = ""
    attachments: Optional[list[Any]] = None

File: rust/crates/rpg-schemas/src/game.rs
//! schemas.game — 游戏核心流程路由请求模型。
use serde::{Deserialize, Serialize};
use serde_json::Value;

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct NewGameRequest {
    pub script_card_id: Option<Value>,
    pub script_id: Option<Value>,
    pub user_card_id: Option<Value>,
    pub persona_id: Option<Value>,
    #[serde(default)]
    pub role: Option<String>,
    #[serde(default = "default_name")]
    pub name: Option<String>,
    #[serde(default)]
    pub background: Option<String>,
}

fn default_name() -> Option<String> { Some("无名者".to_string()) }

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct ChatEstimateRequest {
    #[serde(default)]
    pub message: Option<String>,
    #[serde(default = "default_include_retrieval")]
    pub include_retrieval: Option<bool>,
}

fn default_include_retrieval() -> Option<bool> { Some(true) }

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
pub struct ChatRequest {
    #[serde(default)]
    pub message: Option<String>,
    #[serde(default)]
    pub text: Option<String>,
    pub attachments: Option<Vec<Value>>,
}
